src/dependencies.jl

```diff
@@ -224,6 +224,54 @@ pkg_name(z::Zypper) = z.package
 
 libdir(z::Zypper,dep) = ["/usr/lib", "/usr/lib32", "/usr/lib64"]
 
+# pkg is the system binary package manager for FreeBSD
+const has_bsdpkg = try success(`pkg -v`) catch e false end
+type BSDPkg <: PackageManager
+    package::AbstractString
+end
+can_use(::Type{BSDPkg}) = has_bsdpkg && Sys.KERNEL === :FreeBSD
+function package_available(p::BSDPkg)
+    can_use(BSDPkg) || return false
+    rgx = Regex(string("^(", p.package, ")(\\s+.+)?\$"))
+    for line in eachline(`pkg search -L name $(p.package)`)
+        ismatch(rgx, line) && return true
+    end
+    return false
+end
+function available_version(p::BSDPkg)
+    looknext = false
+    for line in eachline(`pkg search -L name -Q version $(p.package)`)
+        if rstrip(line) == p.package
+            looknext = true
+            continue
+        end
+        if looknext && startswith(line, "Version")
+            # Package versioning is basically a free-for-all, since the package version
+            # is determined entirely by a free text variable in a makefile, though for
+            # the most part things very vaguely resemble semver.
+            rawversion = chomp(line[findfirst(c->c==':', line)+2:end])
+            dashed = replace(rawversion, "_", "-")
+            if ismatch(Base.VERSION_REGEX, rawversion)
+                return VersionNumber(rawversion)
+            elseif ismatch(Base.VERSION_REGEX, dashed)
+                # Many versions are of the form XX.YY.ZZ_A
+                return VersionNumber(dashed)
+            elseif ismatch(r"^\d+\.\d+\.\d+", rawversion)
+                # Good enough
+                return VersionNumber(match(r"^(\d+\.\d+\.\d+)", rawversion).captures[1])
+            elseif ismatch(r"^\d", rawversion)
+                # The version number is so messed up, we're basically just gonna guess
+                return VersionNumber(match(r"^(\d)", rawversion).captures[1])
+            else
```

Comment:
Let me explain the freebsd pkg versioning:
`{orig ver}_{PORTREVISION},{PORTEPOCH}`
e.g. openblas -- `0.2.19_1,1`
The `PORTREVISION` is optional. it increase when any changes made in ports tree.
The `PORTEPOCH` is optional as well. It increase when the software author changing the versioning policy. It make pkg do the version comparison correctly. e.g.: libfoo release version `201705` at first, but later libfoo's versioning becomes `1.2`. In order to determine which one is the newer version, we will check the `PORTEPOCH`.

In our case, I purpose that we can simply remove the `_1,1` at first, then do these matching.